sycl/doc/extensions/proposed/sycl_ext_oneapi_memcpy2d.asciidoc

```diff
@@ -0,0 +1,235 @@
+the implementation supports this feature, or applications can test the macro's
+value to determine which of the extension's features the implementation
+supports.
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|Initial version of this extension.
+|===
+
+=== API Additions
+
+
+The handler class gains the following new methods:
+
+[cols="60a,40"]
+|===
+| Member Function | Description
+
+a|
+[source,c++]
+----
+void ext_oneapi_memcpy2d(void *dest, size_t destPitch,
+  const void *src, size_t srcPitch,
+  size_t width, size_t height)
+----
+
+| Copies the data located at `src` to `dest`. The width of a row in bytes,
+including any padding, is specified by `destPitch` for `dest` and `srcPitch`
+for `src`. The amount of data to copy is specified as `width` bytes per row
+times `height` number of rows.  `width` must not be larger than either
+`srcPitch` or `destPitch`, otherwise this function throws a synchronous
+`exception` with the `errc::invalid` error code. The `dest` and `src`
+parameters must each either be a host pointer or a pointer within a USM
+allocation that is accessible on the handler's device. If a pointer is to a
+USM allocation, that allocation must have been created from the same context
+as the handler's queue.
+
+a|
+[source,c++]
+----
+template <typename T>
+void ext_oneapi_copy2d(const T *src, size_t srcPitch,
+  T *dest, size_t destPitch,
+  size_t width, size_t height)
+----
+
+| Copies the data located at `src` to `dest`. The width of a row in number
+of elements, including any padding, is specified by `destPitch` for `dest`
+and `srcPitch` for `src`. The amount of data to copy is specified as `width`
+elements per row times `height` number of rows. `width` must not be larger
+than either `srcPitch` or `destPitch`, otherwise this function throws a
+synchronous `exception` with the `errc::invalid` error code. The `dest`
+and `src`parameters must each either be a host pointer or a pointer within
+a USM allocation that is accessible on the handler's device. If a pointer is
+to a USM allocation, that allocation must have been created from the same context
+as the handler's queue.
+
+a|
+[source,c++]
+----
+void ext_oneapi_memset2d(void *dest, size_t destPitch,
+  int value, size_t width, size_t height)
+----
+
+| Fills the data located at `dest` with `value`. Note that `value` is
+interpreted as an `unsigned char`. The width of a row in bytes,
+including any padding, is specified by `destPitch` for `dest`.
+The amount of data to fill is specified as `width` bytes per row
+times `height` number of rows.  `width` must not be larger than `destPitch`,
+otherwise this function throws a synchronous `exception` with the
+`errc::invalid` error code. The `dest` parameter must either be a host
+pointer or a pointer within a USM allocation that is accessible on the
+handler's device. If a pointer is to a USM allocation, that allocation
+must have been created from the same context as the handler's queue.
```

Comment:
The core SYCL spec for `handler::memset` requires `dest` to be a USM pointer.  (This is different from the "copy" APIs which allow `src` and `dest` to be either a USM pointer or a host pointer.)  I think the `ext_oneapi_memset2d` should match `handler::memset` in this regard.  The easiest thing is to change this extension spec to require `dest` to be a USM pointer.  If you feel that a host pointer should be allowed, we should file a bug against the core SYCL spec.

Same comment below for `ext_oneapi_fill2d`.